src/forklift/core.py

```diff
@@ -224,12 +224,16 @@ def _hash(crate, hash_path, needs_reproject):
 
     insert_cursor = None
     if needs_reproject:
-        changes.table = arcpy.CreateFeatureclass_management(arcpy.env.scratchGDB,
-                                                            crate.name + reproject_temp_suffix,
-                                                            geometry_type=crate.source_describe.shapeType.upper(),
-                                                            template=crate.source,
-                                                            spatial_reference=crate.source_describe.spatialReference)[0]
-        arcpy.AddField_management(changes.table, src_id_field, 'LONG')
+        changes.table = arcpy.CreateFeatureclass_management(
+            arcpy.env.scratchGDB,
+            crate.name + reproject_temp_suffix,
+            geometry_type=crate.source_describe.shapeType.upper(),
+            template=crate.source,
+            spatial_reference=crate.source_describe.spatialReference)[0]
+        arcpy.AddField_management(changes.table, src_id_field, 'TEXT')
+        if crate.source_describe.dataType == 'ShapeFile':
+            log.info('adding FID field for shapefile comparison')
+            arcpy.AddField_management(changes.table, 'FID', 'LONG')
```

Comment:
Yes, I believe so. `changes.fields` would have FID in it. You could remove this block and run speedtest (which now contains a shapefile) to verify.